textinput_model.go

```diff
@@ -33,133 +33,154 @@ func (p *textinputPlugin) addChar(char []rune) {
 	p.updateEditingState()
 }
 
-func (p *textinputPlugin) MoveCursorHome(modsIsModifier bool, modsIsShift bool, modsIsWordModifierShift bool, modsIsWordModifier bool) {
+func (p *textinputPlugin) moveCursorHomeSimple() {
 	p.selectionBase = 0
-	if !modsIsShift {
-		p.selectionExtent = p.selectionBase
-	}
-	p.updateEditingState()
+	p.selectionExtent = p.selectionBase
 }
 
-func (p *textinputPlugin) MoveCursorEnd(modsIsModifier bool, modsIsShift bool, modsIsWordModifierShift bool, modsIsWordModifier bool) {
+func (p *textinputPlugin) moveCursorHomeSelect() {
+	p.selectionBase = 0
+}
+
+func (p *textinputPlugin) moveCursorEndSimple() {
 	p.selectionBase = len(p.word)
-	if !modsIsShift {
-		p.selectionExtent = p.selectionBase
-	}
-	p.updateEditingState()
+	p.selectionExtent = p.selectionBase
 }
 
-func (p *textinputPlugin) MoveCursorLeft(modsIsModifier bool, modsIsShift bool, modsIsWordModifierShift bool, modsIsWordModifier bool) {
-	if modsIsWordModifierShift {
-		if p.isSelected() {
-			p.selectionExtent = indexStartLeadingWord(p.word, p.selectionExtent)
-		} else {
-			p.selectionExtent = indexStartLeadingWord(p.word, p.selectionBase)
-		}
-	} else if modsIsWordModifier {
-		p.selectionBase = indexStartLeadingWord(p.word, p.selectionBase)
-		p.selectionExtent = p.selectionBase
-	} else if modsIsShift {
+func (p *textinputPlugin) moveCursorEndSelect() {
+	p.selectionBase = len(p.word)
+}
+
+func (p *textinputPlugin) extentSelectionLeftChar() {
+	if p.selectionExtent > 0 {
 		p.selectionExtent--
-	} else if !p.isSelected() {
+	}
+}
+
+func (p *textinputPlugin) extentSelectionLeftWord() {
+	p.selectionBase = indexStartLeadingWord(p.word, p.selectionBase)
+	p.selectionExtent = p.selectionBase
+
+}
+
+func (p *textinputPlugin) extentSelectionLeftLine() {
+	if p.isSelected() {
+		p.selectionExtent = indexStartLeadingWord(p.word, p.selectionExtent)
+	} else {
+		p.selectionExtent = indexStartLeadingWord(p.word, p.selectionBase)
+	}
+
+}
+
+func (p *textinputPlugin) extentSelectionLeftReset() {
+	if !p.isSelected() {
 		if p.selectionBase > 0 {
 			p.selectionBase--
 			p.selectionExtent = p.selectionBase
 		}
 	} else {
 		p.selectionBase = p.selectionExtent
 	}
-	p.updateEditingState()
 }
 
-func (p *textinputPlugin) MoveCursorRight(modsIsModifier bool, modsIsShift bool, modsIsWordModifierShift bool, modsIsWordModifier bool) {
-	if modsIsWordModifierShift {
-		if p.isSelected() {
-			p.selectionExtent = indexEndForwardWord(p.word, p.selectionExtent)
-		} else {
-			p.selectionExtent = indexEndForwardWord(p.word, p.selectionBase)
-		}
-	} else if modsIsWordModifier {
-		p.selectionBase = indexEndForwardWord(p.word, p.selectionBase)
-		p.selectionExtent = p.selectionBase
-	} else if modsIsShift {
+func (p *textinputPlugin) extentSelectionRightChar() {
```

Comment:
The methods should probably be called `extendSelectionRightChar` (etc) with a d, since it's a verb?
The fields `selectionExtent` (etc) are correct, as that is a noun.